hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerStartupCache.java

```diff
@@ -0,0 +1,366 @@
+  private static final String CHECKSUM_PLACEHOLDER = "CHECKSUM_PLACEHOLDER";
+
+  private final ObjectMapper objectMapper;
+
+  /**
+   * Versioning for Container Cache format.
+   */
+  public enum CacheVersion {
+    INITIAL_VERSION(1, "Initial container cache format"),
+
+    FUTURE_VERSION(-1, "Used internally when an unknown cache version is encountered");
+
+    public static final CacheVersion CURRENT = latest();
+    public static final int CURRENT_VERSION = CURRENT.version;
+
+    private final int version;
+    private final String description;
+
+    CacheVersion(int version, String description) {
+      this.version = version;
+      this.description = description;
+    }
+
+    public String description() {
+      return description;
+    }
+
+    public static CacheVersion fromValue(int value) {
+      for (CacheVersion v : values()) {
+        if (v.version == value) {
+          return v;
+        }
+      }
+      return FUTURE_VERSION;
+    }
+
+    private static CacheVersion latest() {
+      CacheVersion[] versions = CacheVersion.values();
+      // Return the latest version (excluding FUTURE_VERSION which is always last)
+      return versions[versions.length - 2];
+    }
+  }
+
+  public ContainerStartupCache() {
+    this.objectMapper = new ObjectMapper();
+    this.objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+  }
+
+  /**
+   * Cache metadata for containers in the volume.
+   */
+  public static class CacheMetadata {
+    private Map<Long, String> containerYamlData;
+    private long rocksDbSequenceNumber;
+    private String checksum;
+    private long timestamp;
+    private int version;
+
+    public CacheMetadata() {
+      this.containerYamlData = new HashMap<>();
+      this.timestamp = System.currentTimeMillis();
+      this.version = CacheVersion.CURRENT_VERSION;
+    }
+
+    public Map<Long, String> getContainerYamlData() {
+      return containerYamlData;
+    }
+
+    public void setContainerYamlData(Map<Long, String> containerYamlData) {
+      this.containerYamlData = containerYamlData;
+    }
+
+    public long getRocksDbSequenceNumber() {
+      return rocksDbSequenceNumber;
+    }
+
+    public void setRocksDbSequenceNumber(long rocksDbSequenceNumber) {
+      this.rocksDbSequenceNumber = rocksDbSequenceNumber;
+    }
+
+    public String getChecksum() {
+      return checksum;
+    }
+
+    public void setChecksum(String checksum) {
+      this.checksum = checksum;
+    }
+
+    public long getTimestamp() {
+      return timestamp;
+    }
+
+    public void setTimestamp(long timestamp) {
+      this.timestamp = timestamp;
+    }
+
+    public int getVersion() {
+      return version;
+    }
+
+    public void setVersion(int version) {
+      this.version = version;
+    }
+  }
+
+  /**
+   * Dump container metadata cache for specific volume.
+   */
+  public File dumpContainerCache(Iterable<? extends Container<?>> containers,
```

Comment:
How about caching in Volume DB itself for any update event in another table ? Not sure about performance for this part.

Reply:
Only the contents of the container's yaml file are cached here, and no content of the DB itself is included. This is because reading the container yaml file is the main bottleneck of startup performance.

It seems that the name of the method here is not very accurate.